src/otx/engine/utils/auto_configurator.py

```diff
@@ -256,41 +254,60 @@ def get_model(self, model_name: str | None = None, label_info: LabelInfo | None
             ...     label_info=<LabelInfo>,
             ... )
         """
+        # TODO(vinnamki): There are some overlaps with src/otx/cli/cli.py::OTXCLI::instantiate_model
         if model_name is not None:
             self._config = self._load_default_config(self.model_name)
-        if label_info is not None:
-            num_classes = label_info.num_classes
-            self.config["model"]["init_args"]["num_classes"] = num_classes
 
-            from otx.core.types.label import HLabelInfo
+        skip = {} if self.task != OTXTaskType.H_LABEL_CLS else {"hlabel_info"}
+
+        model_parser = ArgumentParser()
+        model_parser.add_subclass_arguments(OTXModel, "model", skip=skip, required=False, fail_untyped=False)
 
-            if isinstance(label_info, HLabelInfo):
-                init_args = self.config["model"]["init_args"]
-                init_args["num_multiclass_heads"] = label_info.num_multiclass_heads
-                init_args["num_multilabel_classes"] = label_info.num_multilabel_classes
+        model_config = deepcopy(self.config["model"])
+
+        if label_info is not None and self.task != OTXTaskType.H_LABEL_CLS:
+            model_config["init_args"]["num_classes"] = label_info.num_classes
+        elif label_info is not None and self.task == OTXTaskType.H_LABEL_CLS:
+            model_config["init_args"]["hlabel_info"] = label_info
+        elif label_info is None and self.task == OTXTaskType.H_LABEL_CLS:
+            msg = "You should explicitly give label_info for `OTXTaskType.H_LABEL_CLS` task."
+            raise ValueError(msg)
 
-        logger.warning(f"Set Default Model: {self.config['model']}")
-        return instantiate_class(args=(), init=self.config["model"])
+        return model_parser.instantiate_classes(Namespace(model=model_config)).get("model")
 
     def get_optimizer(self) -> list[OptimizerCallable] | None:
```

Comment:
This seems to be deprecated in `Engine`, do we still need this function?
The same goes for `get_scheduler`.

Reply:
If you really think it as what you said, you can remove this. However, I'll keep this because it is a big sized PR, so that I'd like to avoid any potential test error by removing it.

Reply:
Okay, I see, this is something I'll look into and remove later.